mountpoint-s3/tests/fuse_tests/rmdir_test.rs

```diff
@@ -0,0 +1,83 @@
+use fuser::BackgroundSession;
+use mountpoint_s3::S3FilesystemConfig;
+use std::fs::{self, DirBuilder};
+use tempfile::TempDir;
+use test_case::test_case;
+
+use crate::fuse_tests::{read_dir_to_entry_names, TestClientBox};
+
+fn rmdir_test<F>(creator_fn: F, prefix: &str)
+where
+    F: FnOnce(&str, S3FilesystemConfig) -> (TempDir, BackgroundSession, TestClientBox),
+{
+    let (mount_point, _session, mut test_client) = creator_fn(prefix, Default::default());
+
+    // Create local directory
+    let main_dirname = "test_dir";
+    let main_path = mount_point.path().join(main_dirname);
+    let empty_dirname = "local_empty_dir";
+    let non_empty_dirname = "local_non_empty_dir";
+    let empty_dirpath = main_path.join(empty_dirname);
+    let non_empty_dirpath = main_path.join(non_empty_dirname);
+
+    DirBuilder::new().recursive(true).create(&empty_dirpath).unwrap();
+    DirBuilder::new().recursive(true).create(&non_empty_dirpath).unwrap();
+
+    // verify that the directories exist before testing `rmdir`
+    let read_dir_iter = fs::read_dir(&main_path).unwrap();
+    let dir_entry_names = read_dir_to_entry_names(read_dir_iter);
+    assert_eq!(dir_entry_names, vec![empty_dirname, non_empty_dirname]);
+
+    // Write an object into the directory
+    let filename = "nested_file";
+    let filepath = non_empty_dirpath.join(filename);
+    fs::write(filepath, "").unwrap();
```

Comment:
Should we test that you still can't remove this directory after closing the file too?

Reply:
Ah I see, its this one. https://doc.rust-lang.org/std/fs/fn.write.html

I was thinking it was the lower-level write on a file object. Ignore this comment then :)